api.ts

```diff
@@ -2305,6 +2309,7 @@ export function makeSyncTable<
     execute: wrappedExecute,
     executeUpdate: wrappedExecuteUpdate,
     executeGetPermissions,
```

Comment:
Would we also want to use the same one for `executeUpdate` and `executeGetPermissions`, or leave those be to do their own error handling? If the former, we could either define new properties (`onUpdateError` or `onGetPermissionsError`), or add some parameter to `onError` with the `errorSource`.

Reply:
I'm ok with either of those paths forward, and neither are blocking this PR IIUC